src/sys/socket/addr.rs

```diff
@@ -201,6 +201,28 @@ pub enum AddressFamily {
     Natm = libc::AF_NATM,
 }
 
+impl AddressFamily {
+    /// Create a new `AddressFamily` from an integer value retrieved from `libc`, usually from
+    /// the `sa_family` field of a `sockaddr`.
+    ///
+    /// Currently only supports these address families: Unix, Inet (v4 & v6), Packet, Netlink
+    /// and System. Returns None for unsupported or unknown address families.
+    pub fn from_i32(family: i32) -> Option<AddressFamily> {
```

Comment:
Would it not make more sense for this to be From<i32> instead of a custom method?

Reply:
It only returns an `Option`, so it could be a [`TryFrom`](https://doc.rust-lang.org/std/convert/trait.TryFrom.html) at best. But that's nightly-only...